### PR TITLE
updated joi to version 12

### LINKED ISSUE #38: New joi versions available
Our `package.json` calls for joi 11.x.x.

joi 13.0.0 has been released.

### PR DESCRIPTION
**Pull request by giltho**
Closes #38 
A test is not passing, but since it was not passing before I guess this is normal
That's a small pull request, but well, I saw an issue...
It's an occasion for me to thank you for your work, celebrate is great :smile:

### CODE REVIEW

**Comment by arb** [comment]
Thanks for the PR!

[This issue](https://github.com/hapijs/joi/issues/1326) tells you what changed and why the test failed. The escaping of HTML characters. What do you want to do here @cjihrig? We could fix this and just pass in `escapeHtml` to make the tests pass or some other option I'm not thinking of...

**Comment by cjihrig** [comment]
If possible, I think we should maintain the previous behavior in `celebrate`.

**Comment by arb** [comment]
So @giltho use the `escapeHtml` option when we call `Joi.validate` and then the tests should pass. It should also be possible for users to set that value to `false` if they really want it disabled.

**Comment by giltho** [comment]
Indeed, this should be better, I also added this in the docs, since it is a divergence with Joi

File: README.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ app.use(Celebrate.errors());
 The single exported function from `celebrate`. Returns a `function` with the middleware signature (`(req, res, next)`).
 
 - `schema` - a object where `key` can be one of `'params', 'headers', 'query', and 'body'` and the `value` is a [joi](https://github.com/hapijs/joi/blob/master/API.md) validation schema. Only the `key`s specified will be validated against the incoming `req` object. If you omit a key, that part of the `req` object will not be validated. A schema must contain at least one of the valid keys. 
-- `[options]` - `joi` [options](https://github.com/hapijs/joi/blob/master/API.md#validatevalue-schema-options-callback) that are passed directly into the `validate` function.
+- `[options]` - `joi` [options](https://github.com/hapijs/joi/blob/master/API.md#validatevalue-schema-options-callback) that are passed directly into the `validate` function. Note : `options.escapeHtml` is `true` by default in `Celebrate`, which is not the case for `Joi` since version 12.
```
Comment (cjihrig) [inline README.md]: Small nit - `Note :` -> `Note:`

File: lib/index.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ const Celebrate = (schema, options) => {
   Assert.ifError(result.error);
   const rules = new Map();
 
+  if (options === undefined) {
```
Comment (cjihrig) [inline lib/index.js]: Couldn't this just be:

`options = Object.assign({ escapeHtml: true }, options)`

Comment (giltho) [reply lib/index.js]: Indeed, I don't know why I was pretty sure you could not assign something to undefined, but in fact we can

**Comment by giltho** [comment]
I created a `default` object to which one can easily add properties if it is decided to add more default values that diverge from Joi in the future

**Comment by arb** [comment]
Now that I'm looking at this... we are already escaping html with the `EscapeHtml` library... curious if we need both that lib and the Joi escape HTML?

**Comment by cjihrig** [comment]
No, we probably don't need both.

**Comment by giltho** [comment]
How come the test would not pass when I did not use Joi's option ?

**Comment by giltho** [comment]
Celebrate only uses EscapeHtml to escape paths, which is why the error message that appears in the test was not escaped. However, Joi's escapeHtml option seems to escape some error messages but not paths. It looks like both are not doing the same Job.
I think you can chose to either keep it that way, or escape the whole error with EscapeHtml module.

**Comment by cjihrig** [comment]
Thanks for the PR!

**Pull request merged**